u2f-ref-code/java/src/com/google/u2f/codec/RawMessageCodec.java

```diff
@@ -215,4 +291,65 @@ public static AuthenticateResponse decodeAuthenticateResponse(byte[] data) throw
     .put(challengeSha256);
     return signedData;
   }
+  
+  public static byte[] encodeTransferAccessMessageSignedBytesForAuthenticationKey(
+      byte sequenceNumber, byte[] newUserPublicKey, byte[] applicationSha256,
+      X509Certificate attestationCertificate) throws U2FException {
+
+    byte[] attestationCertificateBytes;
+    try {
+      attestationCertificateBytes = attestationCertificate.getEncoded();
+    } catch (CertificateEncodingException e) {
+      throw new U2FException("Error when encoding attestation certificate.", e);
+    }
+
+    byte[] signedData = new byte[1 + newUserPublicKey.length + applicationSha256.length
+        + attestationCertificateBytes.length];
+    
+    ByteBuffer.wrap(signedData)
```

Comment:
I used byte buffer because that's what is used elsewhere in the code. It also helped me catch a couple missing fields (for example the two extra '+1's for length fields you pointed out in another comment)  but this may not be a good reason to use it. What is the reason that the output stream is better?